### PR TITLE
Value mappings: Add regular-expression based value mapping

### PR DESCRIPTION
**Pull request by mcdee**
This adds a regular-expression based value mapping, which allows replacement of arbitrary text, as per the discussion at https://github.com/grafana/grafana/discussions/37008

Note: the documentation has a link to a graphic.  I couldn't find out where to include the image, pointers as to where it should go would be appreciated.

### CODE REVIEW

**Comment by mckn** [comment]
I think this looks really good! I will add Torkel and Ryan just to get a second pair of eyes since I have not been involved in this feature previously.

**Review by mckn — COMMENTED** [review]
I think this looks really good! I will add Torkel and Ryan just to get a second pair of eyes since I have not been involved in this feature previously.

File: packages/grafana-ui/src/components/ValueMappingsEditor/ValueMappingsEditor.story.tsx [hunk 1]
```diff
@@ -30,6 +30,19 @@ export function Example() {
         },
       },
     },
+    {
+      type: MappingType.RegexToText,
+      options: {
+        // TODO correct format?
```
Comment (ryantxu) [inline packages/grafana-ui/src/components/ValueMappingsEditor/ValueMappingsEditor.story.tsx:36]: is it the correct format? :wink:

**Review by ryantxu — APPROVED** [review]
This looks great!  thank you

File: packages/grafana-ui/src/components/ValueMappingsEditor/ValueMappingsEditor.story.tsx (hunk 1 above)
Comment (mcdee) [reply packages/grafana-ui/src/components/ValueMappingsEditor/ValueMappingsEditor.story.tsx]: As it turned out, yes :smile: 

Have removed the comments.

**Comment by pkolyvas** [comment]
@mcdee Re: your image if you post it here, I can get it to the right place. Our website uses a private repo for the images at the moment.

**Comment by mcdee** [comment]
> @mcdee Re: your image if you post it here, I can get it to the right place. Our website uses a private repo for the images at the moment.

@pkolyvas

**Comment by pkolyvas** [comment]
@mcdee Thanks! On it.

**Pull request merged**